include/ginkgo/core/matrix/csr.hpp

```diff
@@ -979,7 +981,12 @@ class Csr : public EnableLinOp<Csr<ValueType, IndexType>>,
           row_ptrs_(exec, size[0] + 1),
           srow_(exec, strategy->clac_size(num_nonzeros)),
           strategy_(strategy->copy())
-    {}
+    {
+        row_ptrs_.fill(0);
+        col_idxs_.fill(0);
+        values_.fill(zero<value_type>());
```

Comment:
At least row_ptrs needs to be initialized for the matrix to avoid matrices being in a broken state by default.

Otherwise `Csr::create(exec)->apply(b, x)` would be a Segfault Speedrun :smile:

Reply:
I think there we should probably throw if the row_ptrs array is invalid. I think this will degrade performance, as we use this un-initialized creation in a lot of places to declare the objects.

Reply:
That is a significant overhead on SpMV, and may not be possible for sparselib, where it just turns into a segfault.
I can imagine two remedies:
1. add another overload with a tag type "gko::uninitialized" that skips the initialization.
2. use the array constructor and pass uninitialized arrays for the result.